common/workunit/workunit.cpp

```diff
@@ -1023,6 +1023,151 @@ class CConstGraphProgressScopeIterator : public CInterfaceOf<IConstWUScopeIterat
 };
 
 
+int compareNoteScopeOrder(IConstWUException & left, IConstWUException & right)
+{
+    return compareScopeName(left.queryScope(), right.queryScope());
+}
+
+int compareNoteScopeOrder(IInterface * const * left, IInterface * const * right)
+{
+    return compareNoteScopeOrder(*static_cast<IConstWUException *>(*left), *static_cast<IConstWUException *>(*right));
+}
+
+class NotesIterator : public CInterfaceOf<IConstWUScopeIterator>
+{
+public:
+    NotesIterator(const IConstWorkUnit * wu, const ScopeFilter & _filter)
+    {
+        Owned<IConstWUExceptionIterator> exceptions = &wu->getExceptions();
+        ForEach(*exceptions)
+        {
+            IConstWUException & exception = exceptions->query();
+            if (strlen(exception.queryScope())!=0)
```

Comment:
Can queryScope be null?  I think we said that blank scopes would be allowed, but null scopes would be ignored.

Reply:
Apparently, yes.  Fixing.